### ISSUE TITLE
possible to add a coefficient for each collection's field

### ISSUE THREAD

**Issue by piolet** [typesense/typesense#165]
## Description

To improve search result, it is possible to add a coefficient for each collection's field for a better determination of score.

For example, i create a new collection like this :
```
$schema = [
  'name'      => 'companies',
  'fields'    => [
    [
      'name'  => 'company_name',
      'type'  => 'string',
      'coef'  => 2
    ],
    [
      'name'  => 'description',
      'type'  => 'string',
      'coef'  => 1.5
    ],
    [
      'name'  => 'num_employees',
      'type'  => 'int32'
    ],
    [
      'name'  => 'country',
      'type'  => 'string',
      'facet' => true
    ]
  ],
  'default_sorting_field' => 'num_employees'
]
```

if i _search world_ company when `query_by` is on **company_name** and **description**, i could have many result.
thanks to the multiplier coefficient, the score of the results having found _world company_ in the field **company_name** would be higher than those related to the **description**

**Comment by kishorenc** [comment]
👋  This is a duplicate of https://github.com/typesense/typesense/issues/42. 

I haven't yet had a chance to look into this in more detail, but adding custom weights will slow things down during rank score computation on large datasets. It also makes understanding the final result order a bit harder when scores are aggregated across multiple partial matches. I will close this issue but will keep #42 open for further investigation.

**Comment by kishorenc** [comment]
@piolet I forgot to mention that Typesense already respects the order of fields mentioned in the `query_by` parameter. For e.g. if you defined `query_by=title,description`, documents that match on `title` are ranked higher than matches on `description` (all other things being equal).

**Comment by jasonbosco** [comment]
@piolet We just released v0.18.0 with the ability to specify custom weights for fields, using the new query_by_weights parameter: https://github.com/typesense/typesense/releases/tag/v0.18.0.

For eg: `query_by_weights: 1,1,2` with `query_by: field_a,field_b,field_c` will give equal weightage to `field_a` and `field_b`, and will give twice the weightage to `field_c` comparatively.
